model/collection.go

```diff
@@ -0,0 +1,10 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package model
+
+type CollectionMetadata struct {
```

Comment:
Thoughts on adding an API for fetching `TopicMetadata`, with `Id`, `TopicType, `TeamId`, `CollectionType`, `CollectionId`? We still have to figure out what the "preview" requirements will be, but at least this gets us enough that the server can work with the model object when materializing threads.